components/StoryModal.tsx

```diff
@@ -43,17 +43,24 @@ const StoryModal = (): JSX.Element => {
       payload: { description },
     });
     dispatch(editStory(newStory));
+    setDescription('');
   });
 
+  const handleClose = () => {
+    if (description) {
+      saveDescription();
+    }
+    dispatch(deselectStory());
+  };
+
   return (
-    <Modal open={isOpen} key={story?.id} width="60%" onClose={() => dispatch(deselectStory())}>
+    <Modal open={isOpen} key={story?.id} width="60%" onClose={() => handleClose()}>
```

Comment:
This is a good call, we could use this for future edits on the model as well, and save queries to the backend. Nice!